_posts/2020-11-xx-azure-devops-pipeline-as-code.md

```diff
@@ -0,0 +1,234 @@
+---
+layout: post
+title: Upgrade Your Classic Pipelines to Pipeline as Code in Azure DevOps
+author: lassiautio
+excerpt: >
+  Azure DevOps' pipeline as code first sounds a bit scary for those who are familiar with older classic pipeline GUI.
+  This blog post will go through the basics of how to create a build pipeline as code in YAML and gives some tips on where to get help to write YAML pipelines.
+  If you're not yet familiar with YAML pipelines, I consider reading this and try them.
+tags:
+ - Azure DevOps
+ - DevOps
+ - continuous integration
+ - Build pipeline
+ - CI
+---
+
+Azure DevOps used to have only graphical build pipelines, known as Classic Pipeline GUI, which were edited only in the web portal.
+In Build 2019 Microsoft released pipeline as code, where pipelines are defined with YAML (aka YAML pipeline).
+In this blog post, I will go through the basics of YAML pipelines and compare them with the classic pipelines.
+
+## Azure DevOps
+
+Azure DevOps is Microsoft's development, CI/CD, agile, artifact repository, and testing system; "swiss knife" for doing things. It has the following features:
+- Wiki to document things,
+- Boards for agile and scrum teams,
+- Repos to save source code and do pull requests,
+- Pipelines for continuous integration, delivery, and deployment,
+- Test Plans, to run tests and
+- Artifacts to save various kinds of packages (NuGet, NPM, etc.).
+
+Practically it includes most of the things a software team needs except infrastructure to where to run the software.
+Its strength is the smooth integration with all of these features. For example, we can access source code repo from build pipelines, and run build automatically when there is a new commit. Or we can mark tasks on the board as done when its code is deployed via release pipeline.
+
+Even if Azure DevOps is Microsoft's product, it can be used with not-Microsoft programming languages and deploy to any AWS or GCP along with Azure or on-premises. You can use it even if you're not using Microsoft's products.
+
+In this blog post, we will concentrate on pipelines; as code!
```

Comment:
This is a good introduction, giving a bit of an introduction and overview on things, to get everyone on same line in concise manner.